CHANGELOG.md

```diff
@@ -11,6 +11,10 @@ This project adheres to [Semantic Versioning](https://semver.org/).
 
 - Fixed the function signature of `recvmmsg`, potentially causing UB
   ([#2119](https://github.com/nix-rust/nix/issues/2119))
+
+- Fixed the value of optlen as passed to `libc::setsockopt`.
+  (#[#2073](https://github.com/nix-rust/nix/issues/2073))
```

Comment:
This is the wrong PR number.  Also, could you make the message a little bit more clear?  It should include the user-visible change (IpMulticastTtl is fixed) rather than the underlying cause.  The underlying cause is mostly just useful to developers.

Reply:
Originally, I found an issue with `IpMulticastTtl`, but my fix did not address the underlying issue. You identified the underlying issue yourself @asomers and pushed a fix on this PR. So I just rebased on it and added the tests. Thus I also just moved the CHANGELOG.md message that you wrote originally.

The number here points to the issue which I originally raised and you referred to yourself when pushing the fix. I now updated the changelog message in your commit and moved it to the new `changelog` subdir.

Is it fine now? If not, would you prefer that I close this PR and you just push your fix separately and add tests to get your fix in?